mypy/test/testutil.py

```diff
@@ -0,0 +1,83 @@
+from typing import Iterator, List, Tuple
+import time
+import os
+import sys
+import tempfile
+from contextlib import contextmanager
+from threading import Thread
+from unittest import TestCase, main, skipUnless
+from mypy import util
+from mypy.build import random_string
+
+
+WIN32 = sys.platform.startswith("win")
+
+
+def lock_file(filename: str, duration: float) -> Thread:
+    """Open a file and keep it open in a background thread for duration sec."""
+    def _lock_file() -> None:
+        with open(filename):
+            time.sleep(duration)
+    t = Thread(target=_lock_file, daemon=True)
+    t.start()
+    return t
+
+
+@skipUnless(WIN32, "only relevant for Windows")
+class WindowsReplace(TestCase):
+    tmpdir = tempfile.TemporaryDirectory(prefix='mypy-test-',
+                                         dir=os.path.abspath('tmp-test-dirs'))
+    # Choose timeout value that would ensure actual wait inside util.replace is close to timeout
+    timeout = 0.0009 * 2 ** 10
+    short_lock = timeout / 4
+    long_lock = timeout * 2
+
+    threads = []  # type: List[Thread]
```

Comment:
I went back to having 4 tests, since otherwise it's too messy. There's no semantic problem with that, and I'll deal with the extra few seconds in test runtime later.

But with this, I'd like to keep the class-level attributes if you don't see an issue with them; both semantically and performance-wise, I don't want to force each individual test to wait for the threads to expire.